core/Transaction.go

```diff
@@ -0,0 +1,98 @@
+package core
+
+import (
+	"github.com/NethermindEth/juno/core/felt"
+)
+
+// Todo: define contract class
+type ContractClass struct{}
+
+type DeployTransaction struct {
+	// A random number used to distinguish between different instances of the contract.
+	ContractAddressSalt *felt.Felt
+	// The object that defines the contract’s functionality.
+	ContractDef ContractClass
+	// The arguments passed to the constructor during deployment.
+	ConstructorCalldata []*felt.Felt
+	// Who invoked the deployment. Set to 0 (in future: the deploying account contract).
+	CallerAddress *felt.Felt
+	// The transaction’s version. Possible values are 1 or 0.
+	//
+	// When the fields that comprise a transaction change,
+	// either with the addition of a new field or the removal of an existing field,
+	// then the transaction version increases.
+	// Transaction version 0 is deprecated and will be removed in a future version of StarkNet.
+	Version *felt.Felt
+}
+
+func (d *DeployTransaction) Hash() *felt.Felt {
+	// Todo: implement pedersen hash as defined here:
+	// https://docs.starknet.io/documentation/develop/Blocks/transactions/#calculating_the_hash_of_a_deploy_transaction
+	return nil
+}
+
+type InvokeTransaction struct {
+	// Version 0 fields
+	// The address of the contract invoked by this transaction.
+	ContractAddress *felt.Felt
+	// The encoding of the selector for the function invoked (the entry point in the contract)
+	EntryPointSelector *felt.Felt
+
+	// Version 1 fields
+	// The address of the sender of this transaction.
+	SenderAddress *felt.Felt
+	// The transaction nonce.
+	nonce *felt.Felt
```

Comment:
Should the nonce be a public field?
```suggestion
	Nonce *felt.Felt
```